### PR TITLE
Fix in-source builds

### PR DESCRIPTION
**Pull request by madmaxoft**
When configuring, don't create symlinks if link == orig.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -344,30 +344,33 @@ SET_TARGET_PROPERTIES(${CMAKE_PROJECT_NAME} PROPERTIES
...
 	file(TO_NATIVE_PATH "${orig}" orig)
 	file(TO_NATIVE_PATH "${link}" link)
 
-	if (NOT EXISTS ${link})
-		if (CMAKE_HOST_UNIX)
-			set(command ln -s ${orig} ${link})
-		else()
-			if (IS_DIRECTORY ${orig})
-				set(command cmd.exe /c mklink /j ${link} ${orig})
+	if (NOT "${orig}" STREQUAL "${link}")
```
Comment (peterbell10) [inline src/CMakeLists.txt]: This makes the nesting quite deep. Will an early return work instead?
```cmake
if (EXISTS "${link}")
	return()
endif()
```

Comment (madmaxoft) [reply src/CMakeLists.txt]: That's better, I couldn't find if cmake had returns. I'll rework this.

**Comment by madmaxoft** [comment]
The Appveyor times are insane, over an hour for a CI build :(

**Comment by madmaxoft** [comment]
The builds still fail when configured into a folder requiring escaping (such as containing a space). But I'm already in my pajamas.

**Pull request merged**